contributed_definitions/nyaml/NXsource.yaml

```diff
@@ -0,0 +1,521 @@
+category: base
+doc: |
+  The neutron or x-ray storage ring/facility.
+type: group
+NXsource(NXobject):
+  distance(NX_FLOAT):
+    unit: NX_LENGTH
+    doc: |
+      Effective distance from sample
+      Distance as seen by radiation from sample. This number should be negative
+      to signify that it is upstream of the sample.
+  name:
+    doc: |
+      Name of source
+    \@short_name:
+      doc: |
+        short name for source, perhaps the acronym
+  type:
+    doc: |
+      type of radiation source (pick one from the enumerated list and spell exactly)
+    enumeration: [Spallation Neutron Source, Pulsed Reactor Neutron Source, Reactor Neutron Source, Synchrotron X-ray Source, Pulsed Muon Source, Rotating Anode X-ray, Fixed Tube X-ray, UV Laser, Free-Electron Laser, Optical Laser, Ion Source, UV Plasma Source, Metal Jet X-ray]
+  probe:
+    doc: |
+      type of radiation probe (pick one from the enumerated list and spell exactly)
+    enumeration: [neutron, x-ray, muon, electron, ultraviolet, visible light, positron, proton]
```

Comment:
Did we not want to add photon here?

Reply:
I just realized that `NXsource` exists twice now, but in `base_classes` and in `contributed_definitions`. Since it is part of the original, the nyaml file in `contributed_definitions` should be removed.

`base_classes/nyaml/NXsource.yaml` does contain photon here already.

Reply:
I have removed `contributed_definitions/nyaml/NXsource.yaml`. Feel free to resolve.